kymatio/scattering2d/frontend/base_frontend.py

```diff
@@ -23,8 +23,12 @@ def build(self):
             raise RuntimeError('The smallest dimension should be larger than 2^J.')
         self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
         # pads equally on a given side if the amount of padding to add is an even number of pixels, otherwise it adds an extra pixel
-        self.pad = self.backend.Pad([(self.M_padded - self.M) // 2, (self.M_padded - self.M+1) // 2, (self.N_padded - self.N) // 2,
-                                (self.N_padded - self.N + 1) // 2], [self.M, self.N], pre_pad=self.pre_pad)
+        if not self.pre_pad:
+            self.pad = self.backend.Pad([(self.M_padded - self.M) // 2, (self.M_padded - self.M+1) // 2, (self.N_padded - self.N) // 2,
+                                (self.N_padded - self.N + 1) // 2], [self.M, self.N])
+        else:
+            self.pad = lambda x: x
```

Comment:
Do we need this? Can't we just branch inside the scattering function and not fool it into thinking it is padding?

Reply:
I just re-read and learned this is about backwards compatibility